### ISSUE TITLE
Login redirect loop

### ISSUE THREAD

**Issue by th3m4ri0** [activeadmin/activeadmin#4226]
Hi guys,
After some time working on a client website I'm running into an issue in development. When login in, the app generates a redirect loop.
Here's the log: 

```
Started POST "/admin/login" for 127.0.0.1 at 2015-12-06 22:55:01 -0500
Processing by ActiveAdmin::Devise::SessionsController#create as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"yL7S0As5K9AYPyGit2igzcSqoz2FJrGLxxxxxxxx", "user"=>{"email"=>"xxxxxx", "password"=>"[FILTERED]", "remember_me"=>"0"}, "commit"=>"Se connecter"}
  [1m[36mUser Load (0.2ms)[0m  [1mSELECT  "users".* FROM "users"  WHERE "users"."email" = 'xxxxxx'  ORDER BY "users"."id" ASC LIMIT 1[0m
  [1m[35m (0.1ms)[0m  begin transaction
Binary data inserted for `string` type on column `current_sign_in_ip`
Binary data inserted for `string` type on column `last_sign_in_ip`
  [1m[36mSQL (0.5ms)[0m  [1mUPDATE "users" SET "current_sign_in_at" = ?, "current_sign_in_ip" = ?, "last_sign_in_at" = ?, "last_sign_in_ip" = ?, "sign_in_count" = ?, "updated_at" = ? WHERE "users"."id" = 1[0m  [["current_sign_in_at", "2015-12-07 03:55:01.512151"], ["current_sign_in_ip", "127.0.0.1"], ["last_sign_in_at", "2015-12-07 03:55:01.512151"], ["last_sign_in_ip", "127.0.0.1"], ["sign_in_count", 1], ["updated_at", "2015-12-07 03:55:01.513664"]]
  [1m[35m (6.7ms)[0m  commit transaction
Redirected to http://localhost:5000/admin
Completed 302 Found in 107ms (ActiveRecord: 7.5ms)


Started GET "/admin" for 127.0.0.1 at 2015-12-06 22:55:01 -0500
Processing by Admin::DashboardController#index as HTML
  [1m[36mUser Load (0.3ms)[0m  [1mSELECT  "users".* FROM "users"  WHERE "users"."id" = 1  ORDER BY "users"."id" ASC LIMIT 1[0m
Redirected to http://localhost:5000/admin/login
Completed 302 Found in 6ms (ActiveRecord: 0.3ms)


Started GET "/admin/login" for 127.0.0.1 at 2015-12-06 22:55:01 -0500
Processing by ActiveAdmin::Devise::SessionsController#new as HTML
  [1m[35mUser Load (0.2ms)[0m  SELECT  "users".* FROM "users"  WHERE "users"."id" = 1  ORDER BY "users"."id" ASC LIMIT 1
Redirected to http://localhost:5000/admin
Filter chain halted as :require_no_authentication rendered or redirected
Completed 302 Found in 2ms (ActiveRecord: 0.2ms)


Started GET "/admin" for 127.0.0.1 at 2015-12-06 22:55:01 -0500
Processing by Admin::DashboardController#index as HTML
  [1m[36mUser Load (0.2ms)[0m  [1mSELECT  "users".* FROM "users"  WHERE "users"."id" = 1  ORDER BY "users"."id" ASC LIMIT 1[0m
Redirected to http://localhost:5000/admin/login
Completed 302 Found in 2ms (ActiveRecord: 0.2ms)


Started GET "/admin/login" for 127.0.0.1 at 2015-12-06 22:55:01 -0500
Processing by ActiveAdmin::Devise::SessionsController#new as HTML
  [1m[35mUser Load (0.2ms)[0m  SELECT  "users".* FROM "users"  WHERE "users"."id" = 1  ORDER BY "users"."id" ASC LIMIT 1
Redirected to http://localhost:5000/admin
Filter chain halted as :require_no_authentication rendered or redirected
Completed 302 Found in 2ms (ActiveRecord: 0.2ms)


Started GET "/admin" for 127.0.0.1 at 2015-12-06 22:55:01 -0500
Processing by Admin::DashboardController#index as HTML
  [1m[36mUser Load (0.2ms)[0m  [1mSELECT  "users".* FROM "users"  WHERE "users"."id" = 1  ORDER BY "users"."id" ASC LIMIT 1[0m
Redirected to http://localhost:5000/admin/login
Completed 302 Found in 2ms (ActiveRecord: 0.2ms)
```

**Comment by th3m4ri0** [comment]
Wow.
After 2 days of search, I found the problem, and it's not related to Activeadmin.
The loop was due to a Cancancan authorization adapter I added over **5 months ago**, which added a column named `role` to the `users` table. At panel access, the adapter checks for the user role. If it is `==='admin'`, the user is autorized.
For the last 5 months, all went well as I added by hand the role to my test user (locally) and to production users. But I forgot to put it in the seeds. So as I nuked the database to test something, and re populated it with my test users with no role, I introduced this redirection loop.
:sweat: :sweat_smile: 
Case dismissed ! :hammer:
